lib/msal-browser-poc/src/network/XHRClient.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IXhrClient } from "./IXHRClient";
+
+/**
+ * XHR client for JSON endpoints
+ * https://www.npmjs.com/package/async-promise
+ * @hidden
+ */
+export class XhrClient implements IXhrClient {
+    public sendRequestAsync(url: string, requestParams: RequestInit, enableCaching?: boolean): Promise<any> {
+        return new Promise<string>((resolve, reject) => {
+            const xhr = new XMLHttpRequest();
```

Comment:
can we use fetch?

Reply:
This implementation is for the implicit flow, in the auth code library we will switch to using fetch.